modules/puppetca/puppetca_main.rb

```diff
@@ -47,11 +46,12 @@ def disable certname
     def autosign certname
       FileUtils.touch(autosign_file) unless File.exist?(autosign_file)
 
-      autosign = File.open(autosign_file, File::RDWR)
-      # Check that we don't have that host already
-      found = autosign.readlines.find { |line| line.chomp == certname }
-      autosign.puts certname unless found
-      autosign.close
+      open(autosign_file, File::RDWR) do |autosign|
+        # Check that we don't have that host already
+        found = false
+        autosign.each_line { |line| found = true if line.chomp == certname }
```

Comment:
@witlessbird: We have sorted this out. We both agree that the ekohl's push request is fine. Please merge then we can close both requests. Thanks.